docs/affinityassistants.md

```diff
@@ -0,0 +1,116 @@
+<!--
+---
+linkTitle: "Affinity Assistants"
+weight: 405
+---
+-->
+
+# Affinity Assistants
+Affinity Assistant is a feature to coschedule `PipelineRun` `pods` to the same node
+based on [kubernetes pod affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) so that it possible for the taskruns to execute parallel while sharing volume.
+Available Affinity Assistant Modes are **coschedule workspace**, **coschedule pipelineruns**,
+**isolate PipelineRun** and **disabled**. 
+
+* **coschedule workspace** - When a `PersistentVolumeClaim` is used as volume source for a `Workspace` in a `PipelineRun`,
+all `TaskRun` pods within the `PipelineRun` that share the `Workspace` will be scheduled to the same Node. 
+**Note:** Only one pvc-backed workspace can be mounted to each TaskRun in this mode.
+
+* **coschedule pipelineruns (WIP)** - All `TaskRun` pods within the `PipelineRun` will be scheduled to the same Node.
+
+* **isolate PipelineRun (WIP)** - All `TaskRun` pods within the `PipelineRun` will be scheduled to the same Node,
+and only one PipelineRun is allowed to run on a node at a time.
+
+* **disabled** - The Affinity Assistant is disabled. No pod coscheduling behavior.
+
+**Note:** **coschedule pipelineruns** and **isolate PipelineRun** modes are still under development and not yet functional.
+
+This means that Affinity Assistant is incompatible with e.g. other affinity rules
+ configured for the `TaskRun` pods. If the `PipelineRun` has a custom
+[PodTemplate](pipelineruns.md#specifying-a-pod-template) configured, the `NodeSelector` and `Tolerations` fields
+will also be set on the Affinity Assistant pod. The Affinity Assistant
+is deleted when the `PipelineRun` is completed. 
```

Comment:
"Incompatible" is a bit ambiguous here - could you explain this a bit more?

Typically when I think of incompatibility I expect either an error to be returned or one field to take precedence over the other. But in this case it doesn't sound like either is what happens, and Affinity Assistants do coexist with PodTemplates? (though I could be misunderstanding something)

Reply:
I have added some explanation for it, Affinity term in PodTemplate will be overwritten by Affinity Assistant if specified:

https://github.com/tektoncd/pipeline/blob/4f6c74a33b59d2bfc5d5ee14e2a9aa67d74b22f9/pkg/reconciler/taskrun/taskrun.go#L752C1-L756C3